### PR TITLE
Use textlint-rule-no-dead-link, update documents

### LINKED ISSUE #425: docs: use textlint-rule-no-dead-link for linting
[textlint-rule-no-dead-link](https://github.com/textlint-rule/textlint-rule-no-dead-link "textlint-rule-no-dead-link") found dead link in document.

We want to add this rule to our [textlint/docs/.textlintrc](https://github.com/textlint/textlint/blob/master/docs/.textlintrc "textlint/docs/.textlintrc").

## Tasks

- [ ] Add `textlint-rule-no-dead-link` to [textlint/docs/.textlintrc](https://github.com/textlint/textlint/blob/master/docs/.textlintrc "textlint/docs/.textlintrc")
- [ ] Fix dead link that includes #424 

## Related Issue

This rule also can found "docs: Broken link in formatter.md #424"

### PR DESCRIPTION
**Pull request by 0x6b**
Closes #425.

### CODE REVIEW

**Comment by azu** [comment]
Ah, textlint-rule-no-dead-link require Node.js > 6.

**Comment by 0x6b** [comment]
[Travis CI build for Node.js:4](https://travis-ci.org/textlint/textlint/jobs/323536819) failed with following message:

```
error textlint-rule-no-dead-link@4.1.1: The engine "node" is incompatible with this module. Expected version ">=6".
```

Since [no-dead-link](https://github.com/textlint-rule/textlint-rule-no-dead-link) rule supports [node >= 6](https://github.com/textlint-rule/textlint-rule-no-dead-link/blob/739fe5ebc9fce1123d12dd93a6bdd1fecd015bcd/package.json#L6).


[AppVeyor build failure](https://ci.appveyor.com/project/azu/textlint/build/1.0.848) looks interesting:

```
no-dead-link: c:/projects/textlint/examples/config-file is dead. (only absolute urls are supported)
```

Continue reviewing the message.

**Comment by 0x6b** [comment]
"(only absolute urls are supported)" was thrown by [`node-fetch`](https://github.com/bitinn/node-fetch/blob/v1.6.3/index.js#L54).

**Comment by azu** [comment]
The unexpected result is come here

https://github.com/textlint-rule/textlint-rule-no-dead-link/blob/739fe5ebc9fce1123d12dd93a6bdd1fecd015bcd/src/no-dead-link.js#L149
https://github.com/textlint-rule/textlint-rule-no-dead-link/blob/739fe5ebc9fce1123d12dd93a6bdd1fecd015bcd/src/no-dead-link.js#L21

Windows has a protocol that is `c:` .
It is a bug of textlint-rule-no-dead-link.

image

**Comment by 0x6b** [comment]
@azu, thank you for your analysis. I will open an issue for textlint-rule-no-dead-link (it is difficult for me to reproduce/fix this issue locally as I have no Windows environment at my disposal.)

**Comment by 0x6b** [comment]
Opened https://github.com/textlint-rule/textlint-rule-no-dead-link/issues/97.

**Comment by azu** [comment]
I've submit PR to textlint-rule-no-dead-link https://github.com/textlint-rule/textlint-rule-no-dead-link/pull/99

> Since no-dead-link rule supports node >= 6.

We have three options

## 1. Run only on Node.js stable

Linting document is not must on all platform.
We can only run lint on Node.js stable env.

It is realized by matrix testing

- [Customizing the Build - Travis CI](https://docs.travis-ci.com/user/customizing-the-build/ "Customizing the Build - Travis CI")
- https://github.com/textlint/textlint/blob/629caae1859de216eff3b7142623dd98901ab6f6/.travis.yml

## 2. Drop Node.js 4 and use Node.js 6

Node.js 4 will be EOL at April 2018.
I've filed an issue. https://github.com/textlint/textlint/issues/443
We can remove Node.js 4.x from CI before dropping support.

**Comment by 0x6b** [comment]
@azu, thank you for your comments. As you said we don't have to run documentation test on every CI configuration. I prefer option 1 and run only on AppVeyor since I would like to improve documentation going forward but I have no easy access to Windows environment. #434 will made this separation possible. Let me know your thoughts?

**Comment by azu** [comment]
> I prefer option 1

I agree.

FYI: textlint-rule-no-dead-link support windows https://github.com/textlint-rule/textlint-rule-no-dead-link/releases/tag/4.1.2

**Comment by 0x6b** [comment]
@azu, thank you. I will work on option 1. The rule also will have an Node.js v4 support shortly https://github.com/textlint-rule/textlint-rule-no-dead-link/issues/100.

**Comment by 0x6b** [comment]
Reviewing [Travis CI documentation](https://docs.travis-ci.com/user/customizing-the-build/#Build-Matrix).

**Comment by 0x6b** [comment]
Ah, tip of textlint-rule/textlint-rule-no-dead-link support Node.js v4 but have not released yet...

**Comment by azu** [comment]
I've release https://github.com/textlint-rule/textlint-rule-no-dead-link/releases/tag/4.1.3

**Comment by azu** [comment]
FYI: I've invited @0x6b to https://github.com/orgs/textlint-rule/teamœçs/collaborator

**Comment by 0x6b** [comment]
@azu, thank you. I proudly accepted the invitation!

**Comment by 0x6b** [comment]
Finally I have made the CI builds success thanks to help from @azu and @nodaguti. @azu, could you review new Travis CI configuration and documentation updates?

**Comment by azu** [comment]
image
Probably, something is wrong.
(All env is `LINT=true`)

I think that `.travis.yml` can be reverted.
Because, `textlint-rule-no-dead-link` support Node.js v4, Now.

v4 and stable can do same test process.

- v4: lint -> test
- stable: lint -> test

File: .travis.yml [hunk 1]
```diff
@@ -11,4 +14,18 @@ install:
 - yarn install --frozon-lockfile
 - yarn run bootstrap
 script:
-- yarn test:all
```
Comment (azu) [inline .travis.yml]: This change can be reverted by 

```
git revert de1e16799709cba005432d6fa853b8e53b5ae064
```

The combination of matrix and job(beta) seem to do unexpected behavior.

Comment (0x6b) [reply .travis.yml]: @azu, thanks much for your review. How did I miss your point (unexpected `LINT` value)...what I meant to do is:

```diff
@@ -20,13 +20,12 @@ script:
 matrix:
   exclude:
   - node_js: 4
-    env: LINT=true
+    env: LINT=false
   - node_js: stable
-    env: LINT=true
+    env: LINT=false
 jobs:
   include:
     - stage: lint
       if: env(LINT) = true
       script: yarn lint
       node_js: stable
```

which will resulted in:

[My Travis CI build log](https://travis-ci.org/0x6b/textlint/builds/326805622)

Anyway, as you pointed out, we don't have to introduce additional complexity for build. Will revert change to `.travis.yml`, update the PR, run CI build again, then merge.

Comment (azu) [reply .travis.yml]: Year. Thanks for confirm.

> Will revert change to .travis.yml, update the PR, run CI build again, then merge.

OK.

**Comment by 0x6b** [comment]
In summary, what issues this pull request would like to solve:

- Found multiple broken links in documentation (#424)
- Use [textlint-rule-no-dead-link](https://github.com/textlint-rule/textlint-rule-no-dead-link) for linting to avoid similar issue in future (#425)

While working on the pull request, found multiple CI failures:

- AppVeyor: due to the rule had local path resolution issue on Windows (textlint-rule/textlint-rule-no-dead-link#97)
- Travis CI: due to the rule did not support Node.js v4 (textlint-rule/textlint-rule-no-dead-link#100) 

Thanks to help from @azu and @nodaguti and multiple reviews, have two patch releases which resolve CI failures above:

- [4.1.2](https://github.com/textlint-rule/textlint-rule-no-dead-link/releases/tag/4.1.2), Windows support
- [4.1.3](https://github.com/textlint-rule/textlint-rule-no-dead-link/releases/tag/4.1.3), Node.js v4 support

Finally all tests have passed. Will merge this pull request. @azu and @nodaguti, thanks again for your help!

**Comment by azu** [comment]
💯

**Pull request merged**